File: LinkedList/linkedlist.cpp
#include<iostream>
#include<stdlib.h>
using namespace std;

struct node
{
  int data;
  struct node *next;
};

typedef struct node *llist;
llist newNode(int ele)
{
 llist newnode = new node();
 newnode->data = ele;
 newnode->next = NULL;
 return newnode;
}

llist insert(llist list ,int ele)
{
  if(list == NULL)
  {
   list = newNode(ele);
    return list;
  }
  else
  {
   list->next = insert(list->next,ele);
  }
return list;
}

int display(llist list)
{
 llist temp = list;
 int sum = 0;
 while(temp!=NULL)
 {
  cout<<temp->data<<" ";
  sum+=temp->data;
   temp = temp->next;
 }
 cout<<endl;
return sum;
}
void disp (llist list)
{
 while(list!=NULL)
{
  cout<<list->data<<" ";
 list = list->next;
}
cout<<endl;
}
llist reverse(llist list)
{
  llist curr = list;
   llist prev = NULL;
  while(curr)
  {
   llist next = curr->next;
   curr->next = prev;
   prev = curr;
   curr=next;
  }
 return prev;
}
int main()
{
  llist list = NULL;
  list = insert(list,10);
  list = insert(list,13);
  list = insert(list,4);
  list = insert(list,7);
  list = insert(list,17);
 cout<<"Sum of List is : "<<display(list)<<endl;
 disp(list);
 list = reverse(list);
 disp(list);
 cout<<list->next->data<<endl;
  return 0;
}

File: Java/Linkedlist.java
import java.io.*;

class Linkedlist
{
	Node head ;
	class Node
	{
		int data;
		Node next;
		Node(int d)
		{
			data = d;
			next = null;
	        }
	}
       public Node newNode(int d)
	{
		Node new_node = new Node(d);
		return new_node;
	}
	public void push(int d)
	{   
		Node temp = head;
        if(temp == null)
           head = newNode(d);
	else{
          while(temp.next != null)
		  temp = temp.next;
	  temp.next = newNode(d);
	 }
	}

	public void printList()
	{
		Node list = head;
		while(list != null)
		{
			System.out.print(list.data+" ");
			list = list.next;
		}
	   System.out.println("");
	}
	public void middle()
	{
		Node first = head;
		   Node sec = head;
          if(head != null){
	    while(first != null && first.next != null)
	    {
		    first = first.next.next;
		    sec = sec.next;
	    }
	  }
	    System.out.println("middle is "+sec.data);
       }
	


   public static void main(String args[])
   {
	   Linkedlist l = new Linkedlist();
	   for(int i = 1;i<=5;i++)
		l.push(i);
	   l.printList();
	   l.middle();
   }
}


